x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/categorization/CategorizeTextAggregationBuilder.java

```diff
@@ -269,6 +280,16 @@ protected CategorizeTextAggregationBuilder(
 
     @Override
     protected void doWriteTo(StreamOutput out) throws IOException {
+        // Disallow this aggregation in mixed version clusters that cross the algorithm change boundary.
+        if (out.getVersion().before(ALGORITHM_CHANGED_VERSION)) {
```

Comment:
we MAY get this for free with the versioned named writeable stuff. But, keeping this here is cool with me.